src/wire/Socket.ts

```diff
@@ -249,10 +249,14 @@ export class InteractiveSocket extends EventEmitter {
             return;
         }
 
-        this.state = SocketState.Closing;
-        this.socket.close(1000, 'Closed normally.');
-        this.queue.forEach(packet => packet.cancel());
-        this.queue.clear();
+        if (this.state !== SocketState.Idle) {
+            this.state = SocketState.Closing;
+            this.socket.close(1000, 'Closed normally.');
+            this.queue.forEach(packet => packet.cancel());
+            this.queue.clear();
+        } else {
+            this.state = SocketState.Closing;
```

Comment:
there's no 'Closed' state on the SocketState.

Idle, Connecting, Connected, Closing, Reconnecting, Refreshing

Happy to add a Closed state though :)